src/ui/Misc/FileTypes.ts

```diff
@@ -70,9 +75,13 @@ export class FileTypes {
       // The main dictionary by using the original value.
       localizedString = l(filetype);
     }
+
+    const iconClass = StringUtils.htmlEncode(loweredCaseFiletype).replace(' ', '-');
+    const caption = StringUtils.htmlEncode(localizedString);
```

Comment:
Since these lines are repeated above, could we extract them into a shared method?
It would also be [safer](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md#why-cant-i-just-html-entity-encode-untrusted-data) to escape the characters using underscore's method rather than encoding them.

As Kyu mentioned, unit tests are helpful make sure we never reintroduce the issue.